teachertool/src/transforms/readdCriteriaToChecklist.ts

```diff
@@ -0,0 +1,28 @@
+import { stateAndDispatch } from "../state";
+import { logDebug } from "../services/loggingService";
+import { setChecklist } from "./setChecklist";
+import { Ticks } from "../constants";
+import { getCriteriaInstanceWithId } from "../state/helpers";
+
+export function readdCriteriaToChecklist(criteriaInstanceId: string) {
```

Comment:
Over in `addCriteriaToChecklist` do you want to look for an existing deleted matching criteria and restore it if found? Would that make the distinction between adding and re-adding unnecessary?

Reply:
An alternative might be to permanently delete the criteria instance once the toast disappears.

Reply:
I think permanently deleting is probably for the best with this approach. Though see https://github.com/microsoft/pxt/pull/10165#discussion_r1752355733 for some thoughts on a slightly different approach that I think would lessen the importance of it (would still ultimately be good to have, though).

Reply:
Just for clarity, we are now permanently deleting the criteria after the toast goes away. I also wanted to add since this is related to the `max` problem that @eanders-ms expressed above, I removed the check for whether a criteria is deleted on the `CatalogOverlay`. I did so because having the check would allow a user to add and also readd a criteria, thus possibly having two instances of a criteria that is supposed to only have one. Getting rid of the check in the overlay makes it so it still looks like I can't add the "one only" criteria until the criteria is permanently deleted, thus avoiding the ability for a user to add and readd a criteria that can only be used once.